Automattic/jetpack
# Add/my jetpack plan refresh

Adds the endpoint to refresh plan data which can be called from wpcom when the advanced plan is purchased. 

This PR is branched off https://github.com/Automattic/jetpack/pull/29529, so merge this one after the linked PR is merged. 
## Proposed changes:

* Added the endpoint
* Also refresh the current local plan data whenever we fetch the plans in the my jetpack page. 

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:





On both the jetpack and jetpack social plugins, while the other is inactive (i.e social is active, jetpack is inactive and vice versa)
* Sandbox D105413-code
* Remove the jetpack social advanced plan from your site and go to the my jetpack page to make sure your plan data store on your jetpack site is upto date. 
* Buy the Jetpack Social Advanced plan on your site and go to the editor to make sure that the native posting options show up.
* Remove the plan from mc and go to the my jetpack page. 
* You will see the prompt to buy the paid plans in the editor sidebar.
